components/script/stylesheet_loader.rs

```diff
@@ -85,9 +85,21 @@ pub struct StylesheetContext {
     /// The node document for elem when the load was initiated.
     document: Trusted<Document>,
     origin_clean: bool,
+    /// A token which must match the generation id of the `HTMLLinkElement` for it to load the stylesheet.
+    generation_id: GenerationId,
 }
 
-impl PreInvoke for StylesheetContext {}
+impl PreInvoke for StylesheetContext {
+    fn should_invoke(&self) -> bool {
+        // We must first check whether the generations of the context and the element match up,
+        // else we risk applying the wrong stylesheet when responses come out-of-order.
+        let gen = self.elem.root()
```

Comment:
We should invoke the callback always, just not set the stylesheet if the generation doesn't match. Otherwise we will lose track of the pending load count, which means we'll never fire the load event.